src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1991,7 +2004,11 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
                             helperId = ReadyToRunHelperId.GetThreadStaticBase;
 #endif
                         }
-                        else if (field.HasGCStaticBase)
+                        else if (field.HasGCStaticBase
+#if READYTORUN
+                            || isGCBoxedStaticField
```

Comment:
Well, I think this is just caused by the fact that statics of [non-GC] struct types are stored as internally boxed in CoreCLR; there is a special JIT flag that must be set when accessing them so that it properly traverses the box to reach the actual content of the static. So I believe it's this implicit box, not the static itself, that is a GC object. I apologize if I missed your point.

Reply:
```csharp
struct Foo
{
    int i;
}

class Program
{
    static object s_o;
    static Foo s_f;
}
```

When accessing `Program.s_f`, is this going to tell RyuJIT to use the `GetGCStaticBase` helper (`isGCBoxedStaticField` is going to be true because it's a non-primitive value type)?

The `s_f` field is going to be placed in the non-GC static section AFAIK (according to what it's told by `ReadyToRunCompilerContext.ComputeHasGCStaticBase`). The `offset` we report to RyuJIT is going to be the offset calculated by the type system (it's going to be the same offset as the `s_o` field because they have different bases).

Reply:
The code moved around in the last commit, but the suspicious "report GC bases to the JIT that don't match how the type system laid out the type" flag is still here.

I'm pretty sure that for the example I gave you above, this is going to say "use the `GetGCStaticBase` helper and access offset 0 from it" for both the `s_o` and `s_f` fields. The type system actually laid out `s_f` on the non-GC static base.